test/api/route/typing_test.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:checks/checks.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/api/route/messages.dart';
+import 'package:zulip/api/route/typing.dart';
+
+import '../../stdlib_checks.dart';
+import '../fake_api.dart';
+
+void main() {
+  const streamId = 123;
+  const topic = 'topic';
+  const userIds = [101, 102, 103];
+
+  Future<void> checkSetTypingStatus(FakeApiConnection connection, TypingOp op, {
+    required MessageDestination destination,
+    required Map<String, String> expectedBodyFields,
+  }) async {
+    connection.prepare(json: {});
+    await setTypingStatus(connection, op: op, destination: destination);
+    check(connection.lastRequest).isA<http.Request>()
+      ..method.equals('POST')
+      ..url.path.equals('/api/v1/typing')
+      ..bodyFields.deepEquals(expectedBodyFields);
+  }
+
+  Future<void> checkSetTypingStatusForTopic(TypingOp op, String expectedOp) =>
+    FakeApiConnection.with_((connection) =>
+      checkSetTypingStatus(connection, op,
```

Comment:
nit: block bodies

see that Flutter style guide section I've linked a few times before :slightly_smiling_face: